indico/modules/rb/client/js/modules/admin/MapAreasPage.jsx

```diff
@@ -0,0 +1,356 @@
+            return null;
+        }
+
+        const {areas} = this.props;
+        return areas.map((area) => {
+            const southWest = [area.top_left_latitude, area.top_left_longitude];
+            const northEast = [area.bottom_right_latitude, area.bottom_right_longitude];
+            const latLngBounds = new Leaflet.LatLngBounds([southWest, northEast]);
+            const center = this.map.leafletElement.latLngToContainerPoint(latLngBounds.getCenter());
+            const bottomRight = this.map.leafletElement.latLngToContainerPoint(latLngBounds.getSouthEast());
+            const onAreaClick = () => {
+                const {suppressClick} = this.state;
+                if (suppressClick) {
+                    return;
+                }
+
+                this.setState({
+                    areaProps: {
+                        bounds: {
+                            southWest: latLngBounds.getSouthWest(),
+                            northEast: latLngBounds.getNorthEast()
+                        },
+                        id: area.id,
+                        isDefault: area.is_default,
+                        name: area.name
+                    }
+                });
+            };
+
+            return (
+                <Rectangle key={`${area.name}-${center.x}-${center.y}`}
+                           bounds={[southWest, northEast]}
+                           name={area.name}
+                           isDefault={area.is_default}
+                           onClick={onAreaClick}
+                           id={area.id}>
+                    <Tooltip offset={[bottomRight.x - center.x, bottomRight.y - center.y]}
+                             direction="bottom"
+                             sticky
+                             permanent
+                             interactive>
+                        <>
+                            {area.name}
+                            {area.is_default && (
+                                <span className="area-default">★</span>
+                            )}
+                        </>
+                    </Tooltip>
+                </Rectangle>
+            );
+        });
+    };
+
+    onSubmit = async (data) => {
+        const {actions: {createMapArea, updateMapAreas}} = this.props;
+        const {areaProps: {id}} = this.state;
+
+        if (!id) {
+            await createMapArea(data);
+        } else {
+            await updateMapAreas([{...data, id}]);
+        }
+
+        this.setState({areaProps: null});
+    };
+
+    renderModalForm = () => {
+        const {areaProps} = this.state;
+        if (!areaProps) {
+            return null;
+        }
+
+        const {id, bounds, name, isDefault} = areaProps;
+        return (
+            <FinalForm onSubmit={this.onSubmit}
+                       render={this.renderForm}
+                       subscription={{}}
+                       initialValues={{
+                           name: !id ? null : name,
```

Comment:
`id === null` (or `undefined`) - while there shouldn't be MapAreas with id 0, I'd rather do an explicit check for null/undefined here.